docs/whirlpool/docs/03-Whirlpools SDKs/01-Whirlpools/06-Send Transaction.mdx

```diff
@@ -0,0 +1,372 @@
+        rpc
+      });
+    const computeUnitEstimate = await getComputeUnitEstimateForTransactionMessage(transactionMessage) + 100_000;
+    const medianPrioritizationFee = await rpc.getRecentPrioritizationFees()
+      .send()
+      .then(fees => fees[Math.floor(fees.length / 2)].prioritizationFee)
+    const transactionMessageWithComputeUnitInstructions = await prependTransactionMessageInstructions([
+      getSetComputeUnitLimitInstruction({ units: computeUnitEstimate }),
+      getSetComputeUnitPriceInstruction({ microLamports: medianPrioritizationFee })
+    ], transactionMessage);
+    ```
+  </TabItem>
+  <TabItem value="rust" label="Rust">
+    ```rust title="main.rs"
+    #[tokio::main]
+    async fn main() {
+      // ...
+      let simulated_transaction = rpc.simulate_transaction(&transaction).await.unwrap();
+
+      let mut all_instructions = vec![];
+      if let Some(units_consumed) = simulated_transaction.value.units_consumed {
+        let units_consumed_safe = units_consumed as u32 + 100_000;
+        let compute_limit_instruction =
+          ComputeBudgetInstruction::set_compute_unit_limit(units_consumed_safe);
+        all_instructions.push(compute_limit_instruction);
+
+        let prioritization_fees = rpc
+          .get_recent_prioritization_fees(&[whirlpool_address])
+          .await
+          .unwrap();
+        let mut prioritization_fees_array: Vec<u64> = prioritization_fees
+          .iter()
+          .map(|fee| fee.prioritization_fee)
+          .collect();
+        prioritization_fees_array.sort_unstable();
+        let prioritization_fee = prioritization_fees_array
+          .get(prioritization_fees_array.len() / 2)
+          .cloned();
+
+        if let Some(prioritization_fee) = prioritization_fee {
+          let priority_fee_instruction =
+            ComputeBudgetInstruction::set_compute_unit_price(prioritization_fee);
+          all_instructions.push(priority_fee_instruction);
+        }
+      }
+      // ...
+    }
+    ```
+  </TabItem>
+</Tabs>
+
+### 6: Sign and Submit Transaction
+Finally, the transaction is signed, encoded, and submitted to the network. A client-side retry mechanism ensures that the transaction is repeatedly sent until it is confirmed or the blockhash expires. The same signed transaction is sent in a loop with retries at fixed intervals (100ms), while polling the status using getSignatureStatuses.
+
+<Tabs groupId="programming-languages">
+  <TabItem value="ts" label="Typescript" default>
+    ```tsx title="sendTransaction.ts"
+    const signedTransaction = await signTransactionMessageWithSigners(transactionMessageWithComputeUnitInstructions)
+    const base64EncodedWireTransaction = getBase64EncodedWireTransaction(signedTransaction);
+
+    let blockHashLifetime = 150n;
+    let currentBlockHeight = latestBlockHash.value.lastValidBlockHeight - blockHashLifetime;
+    while (latestBlockHash.value.lastValidBlockHeight >= currentBlockHeight) {
```

Comment:
I think in most places we just have a time-based loop. This is advantageous if you send to multiple rpc nodes because some might be running behind more than others (meaning that `rpc.getBlockHeight().send()` is never the actual blockheight of the chain)

Reply:
Yeah, I know this. My thinking was that this approach shows how blockhashes work, but it makes more sense to give the example with the time-based loop